.github/workflows/python-CI.yml

```diff
@@ -105,21 +105,104 @@ jobs:
         run: |
           python -m pip install --upgrade pip==${{ env.pip-version }}
           pip install hatch
-      - name: Proto codegen
+      - name: Check notebooks have cleaned output and metadata
         run: |
-          hatch run proto:recompile
+          hatch run notebooks:clean
           git diff --exit-code
-      - name: Gql codegen
+
+  build-graphql-schema:
+    name: Build GraphQL Schema
+    needs: changes
+    if: ${{ needs.changes.outputs.phoenix == 'true' }}
+    strategy:
+      matrix:
+        py: [ 3.12 ]
+        os: [ ubuntu-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.py }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.py }}
+      - name: Set up `uv`
+        uses: astral-sh/setup-uv@v3
+        with:
+          version: 0.4.22
+          enable-cache: true
+          cache-dependency-glob: |
+            pyproject.toml
+            requirements/build-graphql-schema.txt
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up `tox` with `tox-uv`
+        run: uv tool install tox --with tox-uv
+      - name: Build Schema
         run: |
-          hatch run gql:build
+          tox run -e build_graphql_schema
           git diff --exit-code
-      - name: Export OpenAPI schema
+
+  build-openapi-schema:
+    name: Build OpenAPI Schema
+    needs: changes
+    if: ${{ needs.changes.outputs.phoenix == 'true' }}
+    strategy:
+      matrix:
+        py: [ 3.12 ]
+        os: [ ubuntu-latest ]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+      - name: Set up Python ${{ matrix.py }}
+        uses: actions/setup-python@v5
+        with:
+          python-version: ${{ matrix.py }}
+      - name: Set up `uv`
+        uses: astral-sh/setup-uv@v3
+        with:
+          version: 0.4.22
+          enable-cache: true
+          cache-dependency-glob: |
+            pyproject.toml
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set up `tox` with `tox-uv`
+        run: uv tool install tox --with tox-uv
+      - name: Build OpenAPI Schema
         run: |
-          hatch run openapi:build
+          tox run -e build_openapi_schema
           git diff --exit-code
-      - name: Check notebooks have cleaned output and metadata
+
+  compile-protobuf:
+    name: Compile Protobuf
+    needs: changes
+    if: ${{ needs.changes.outputs.phoenix == 'true' }}
+    strategy:
+      matrix:
+        py: [ 3.11 ]
```

Comment:
Using 3.11 to make sure it runs in the container?

Reply:
oh i added a comment after your comment. uv is failing for unknown reasons